### PR TITLE
=doc #24435 copy mutable maps before sharing them as messages in java

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/24435

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java [hunk 1]
```diff
@@ -139,8 +139,14 @@ private void onTerminated(Terminated t) {
...
   private void onAllTemperatures(RequestAllTemperatures r) {
+    // since Java collections are mutable, we want to avoid sharing them between actors (since multiple Actors (threads)
+    // modifying the same mutable data-structure is not safe), and perform a defensive copy of the mutable map:
+    //
+    // Feel free to use your favourite immutable data-structures library with Akka in Java applications!
+    HashMap<ActorRef, String> copyActorToDeviceId = new HashMap<>(this.actorToDeviceId);
+
     getContext().actorOf(DeviceGroupQuery.props(
-            actorToDeviceId, r.requestId, getSender(), new FiniteDuration(3, TimeUnit.SECONDS)));
+        copyActorToDeviceId, r.requestId, getSender(), new FiniteDuration(3, TimeUnit.SECONDS)));
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java]: Now it sounds like a method, wouldn't `actorToDeviceIdCopy` be better?

Comment (ktoso) [reply akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java]: Good point, thanks

**Comment by ktoso** [comment]
Addressed feedback

**Review by johanandren — APPROVED** [review]
Thanks, LGTM!

File: akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java [hunk 2]
```diff
@@ -139,8 +139,14 @@ private void onTerminated(Terminated t) {
   //#query-added
 
   private void onAllTemperatures(RequestAllTemperatures r) {
+    // since Java collections are mutable, we want to avoid sharing them between actors (since multiple Actors (threads)
+    // modifying the same mutable data-structure is not safe), and perform a defensive copy of the mutable map:
+    //
+    // Feel free to use your favourite immutable data-structures library with Akka in Java applications!
+    HashMap<ActorRef, String> actorToDeviceIdCopy = new HashMap<>(this.actorToDeviceId);
```
Comment (chbatey) [inline akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java]: If this is in the docs i'd type this to Map

Comment (chbatey) [reply akka-docs/src/test/java/jdocs/tutorial_5/DeviceGroup.java]: java makes me sad

File: akka-docs/src/test/java/jdocs/tutorial_6/DeviceGroup.java [hunk 3]
```diff
@@ -133,8 +133,14 @@ private void onTerminated(Terminated t) {
   }
 
   private void onAllTemperatures(RequestAllTemperatures r) {
+    // since Java collections are mutable, we want to avoid sharing them between actors (since multiple Actors (threads)
+    // modifying the same mutable data-structure is not safe), and perform a defensive copy of the mutable map:
+    //
+    // Feel free to use your favourite immutable data-structures library with Akka in Java applications!
+    HashMap<ActorRef, String> actorToDeviceIdCopy = new HashMap<>(this.actorToDeviceId);
```
Comment (chbatey) [inline akka-docs/src/test/java/jdocs/tutorial_6/DeviceGroup.java]: Map

**Review by chbatey — APPROVED** [review]
One nit then LGTM

**Comment by ktoso** [comment]
Thanks for reviews merging

**Pull request merged**